### PR TITLE
Python3: make newly introduced python file Python3 compatible

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

File: fetch/connection-pool/resources/network-partition-key.py [hunk 1]
```diff
@@ -85,29 +83,29 @@ def handle_fetch_file(request, response, partition_id, uuid):
...
 
-    file = open(path, mode="r")
+    file = open(path, mode="rb")
     body = file.read()
     file.close()
 
-    subresource_path = "/" + os.path.relpath(__file__, base_path).replace('\\','/')
-    subresource_params = "?partition_id=" + partition_id + "&uuid=" + uuid + "&subresource_origin=" + subresource_origin + "&include_credentials=" + include_credentials
-    body = body.replace("SUBRESOURCE_PREFIX:", subresource_origin + subresource_path + subresource_params)
+    subresource_path = u"/" + os.path.relpath(isomorphic_decode(__file__), base_path).replace(u'\\', u'/')
```
Comment (Hexcles) [inline fetch/connection-pool/resources/network-partition-key.py:102]: Nit: it's better to make `subresource_path` binary in the first place, as all subsequent uses (along with `subresource_params`) are all binary.

File: fetch/connection-pool/resources/network-partition-key.py [hunk 2]
```diff
@@ -27,17 +25,17 @@ def main(request, response):
...
-            server_state = stash.take(uuid) or { "test_failed": False }
-            if address_key in server_state and server_state[address_key] != partition_id:
-                server_state["test_failed"] = True
-            server_state[address_key] = partition_id
-            test_failed = server_state["test_failed"]
+            address_key = str(request.client_address) + u"|" + str(request.url_parts.port)
+            server_state = stash.take(uuid) or {u"test_failed": False}
+            if address_key in server_state and server_state[address_key] != isomorphic_decode(partition_id):
+                server_state[u"test_failed"] = True
+            server_state[address_key] = isomorphic_decode(partition_id)
```
Comment (Hexcles) [inline fetch/connection-pool/resources/network-partition-key.py:37]: Nit: can we directly store and compare the binary `partition_id`?

**Review by Hexcles — APPROVED** [review]
LGTM % nits

File: fetch/connection-pool/resources/network-partition-key.py (hunk 1 above)
Comment (ziransun) [reply fetch/connection-pool/resources/network-partition-key.py]: Code updated. Thank you!

File: fetch/connection-pool/resources/network-partition-key.py (hunk 2 above)
Comment (ziransun) [reply fetch/connection-pool/resources/network-partition-key.py]: Yes. Code updated.

**Comment by ziransun** [comment]
@Hexcles, are the updated changes okay? Thank you!

**Pull request merged**